apps/frontend/core-dapp/src/features/trade/TradeStore.ts

```diff
@@ -618,4 +649,36 @@ export class TradeStore {
 
     return +this.openTradeAmount / amountOutAfterSlippage
   }
+
+  get closePositionValue(): string {
+    if (this.selectedMarket?.resolved) {
+      return this.selectedPosition?.totalValue ?? ''
+    }
+
+    return this.userClosePositionValue
+  }
+
+  private get redeemPositionAmountOutBN(): BigNumber | undefined {
+    const { closePositionAmountBN, selectedMarket, direction } = this
+    if (selectedMarket === undefined) return undefined
+    const { finalLongPrice, maxPrice } = selectedMarket
+
+    if (
+      finalLongPrice === undefined ||
+      maxPrice === undefined ||
+      closePositionAmountBN === undefined
+    ) {
+      return undefined
+    }
+
+    if (direction === 'long') {
+      return closePositionAmountBN.mul(finalLongPrice).div(maxPrice)
+    }
+
+    return closePositionAmountBN.mul(maxPrice.sub(finalLongPrice)).div(maxPrice)
+  }
```

Comment:
Maybe we could just use user's balance here instead of relying on closePositionAmountBN? You could use the `this.selectedPosition.token.balanceOfSigner` which will select the direction for you so you don't have to do `direction === 'long'`

```suggestion
    if (
      finalLongPrice === undefined ||
      maxPrice === undefined ||
      this.selectedPosition.token.balanceOfSigner === undefined
    ) {
      return undefined
    }

    return this.selectedPosition.token.balanceOfSigner.mul(maxPrice.sub(finalLongPrice)).div(maxPrice)
  }
```

Reply:
We still need the direction because the formula is different. The amount of collateral you get for your position is:

* For long positions: `long tokens * final long payout %`
* For short positions: `short tokens * (1 - final long payout %)`

But yeah, I agree that relying on `selectedPosition.token.balanceOfSigner` is simpler and clearer than on `closePositionAmoutnBN`